src/daemon/rpc_command_executor.cpp

```diff
@@ -1992,4 +1996,341 @@ bool t_rpc_command_executor::get_service_node_registration_cmd(const std::vector
     tools::success_msg_writer() << res.registration_cmd;
     return true;
 }
+
+// Returns lowest x such that (STAKING_PORTIONS * x/amount) >= portions
+static uint64_t get_amount_to_make_portions(uint64_t amount, uint32_t portions)
+{
+  uint64_t lo, hi, resulthi, resultlo;
+  lo = mul128(amount, portions, &hi);
+  if (lo > UINT64_MAX - (STAKING_PORTIONS - 1))
+    hi++;
+  lo += STAKING_PORTIONS-1;
+  div128_32(hi, lo, STAKING_PORTIONS, &resulthi, &resultlo);
+  return resultlo;
+}
+static uint64_t get_actual_amount(uint64_t amount, uint32_t portions)
+{
+  uint64_t lo, hi, resulthi, resultlo;
+  lo = mul128(amount, portions, &hi);
+  div128_32(hi, lo, STAKING_PORTIONS, &resulthi, &resultlo);
+  return resultlo;
+}
+bool t_rpc_command_executor::prepare_registration()
+{
+  std::string categories = mlog_get_categories();
+  mlog_set_categories("");
+
+  cryptonote::COMMAND_RPC_GET_INFO::request req;
+  cryptonote::COMMAND_RPC_GET_INFO::response res;
+
+  if (m_is_rpc)
+  {
+    std::cout << "Cannot prepare registration over RPC" << std::endl;
+    mlog_set_categories(categories.c_str());
+    return true;
+  }
+  // TODO: ensure the block height is up to date?
+  if (!m_rpc_server->on_get_info(req, res) || res.status != CORE_RPC_STATUS_OK)
+  {
+    std::cout << "Could not get current blockchain info" << std::endl;
+    mlog_set_categories(categories.c_str());
+    return true;
+  }
+
+#ifdef HAVE_READLINE
+  rdln::suspend_readline pause_readline;
+#endif
+
+  size_t number_participants = 1;
+  uint32_t operating_cost_portions = STAKING_PORTIONS;
+  bool is_solo_stake = false;
+  const auto nettype = m_rpc_server->nettype();
+
+  const uint64_t DUST = 10; // anything less than 10 ROK is considered dust and will be added to operator stake
```

Comment:
Should be (staking requirement / staking portions + 1) * (max num contributors - 1)

Reply:
Which is at most 0.000031434 loki